src/java/org/apache/cassandra/gms/EndpointState.java

```diff
@@ -138,6 +143,12 @@ void markDead()
         isAlive = false;
     }
 
+    public boolean isEmpty()
```

Comment:
I have a slight preference to renaming this to `isEmptyWithoutStatus` or better name to prevent potential misuse, the state may still contain other non-status fields.

Reply:
rebased (to resolve conflicts with trunk) and renamed; the second commit is only this rename.